File: solutions/cpp/protein-translation/1/protein_translation.cpp
#include "protein_translation.h"
using namespace std;
namespace protein_translation {
    vector<string> proteins(string rna){
        map<string,string> rtop = {
            {"AUG",	"Methionine"},
            {"UUU", "Phenylalanine"},
            {"UUC", "Phenylalanine"},
            {"UUA", "Leucine"},
            {"UUG", "Leucine"},
            {"UCU", "Serine"},
            {"UCC", "Serine"},
            {"UCA", "Serine"},
            {"UCG", "Serine"},
            {"UAU", "Tyrosine"},
            {"UAC", "Tyrosine"},
            {"UGU", "Cysteine"},
            {"UGC", "Cysteine"},
            {"UGG", "Tryptophan"},
            {"UAA", "STOP"},
            {"UAG", "STOP"},
            {"UGA", "STOP"}
        };

        vector<string> prots;
        for (size_t i=0; i<rna.length(); i+=3){
            string codon = rna.substr(i,3);
            string p = rtop[codon];
            if (p=="STOP") return prots;
            else prots.push_back(p);
        }
        return prots;

    }
}  // namespace protein_translation

File: solutions/csharp/protein-translation/1/ProteinTranslation.cs
using System;
using System.Collections.Generic;

public static class ProteinTranslation
{
    public static string[] Proteins(string strand)
    {
        List<string> prot = new List<string> ();
        for (int i=0;i<strand.Length;i+=3){
            switch (strand.Substring(i,3)){
                case "AUG": prot.Add("Methionine"); break;
                case "UUU": case "UUC" : prot.Add(  "Phenylalanine"); break;
                case "UUA": case "UUG": prot.Add( 	"Leucine"); break;
                case "UCU": case "UCC": case "UCA": case "UCG":	prot.Add("Serine"); break;
                case "UAU": case "UAC": prot.Add(  	"Tyrosine"); break;
                case "UGU": case "UGC": prot.Add( "Cysteine"); break;
                case "UGG": prot.Add("Tryptophan"); break;
                case "UAA": case "UAG": case "UGA":	return prot.ToArray();
            }
        }
        return prot.ToArray();
    }
}